lib/explorer/data_frame.ex

```diff
@@ -3019,7 +3019,13 @@ defmodule Explorer.DataFrame do
   defp put(%DataFrame{} = df, column_name, fun, arg, opts) when is_column_name(column_name) do
     opts = Keyword.validate!(opts, [:dtype])
     name = to_column_name(column_name)
-    dtype = opts[:dtype]
+
+    dtype =
+      if opts[:dtype] in [nil, :infer] do
+        opts[:dtype]
+      else
+        Shared.normalise_dtype!(opts[:dtype])
+      end
```

Comment:
We are checking for dtype being nil or infer in 3 diofferent places now. Maybe we can move this normalization to line 3039 (a couple lines below?).

Reply:
Something like this? 94daa48